phoenix-ios/phoenix-ios/views/HomeView.swift

```diff
@@ -643,54 +643,144 @@ fileprivate struct PaymentCell : View, ViewName {
 	}
 }
 
-fileprivate struct ConnectionStatusButton : View {
+fileprivate struct AppStatusButton: View, ViewName {
 	
 	@State var dimStatus = false
+	
+	@State var syncState: SyncManagerState = .initializing
+	@State var pendingSettings: PendingSettings? = nil
+	
 	@StateObject var connectionsMonitor = ObservableConnectionsMonitor()
 	
 	@Environment(\.popoverState) var popoverState: PopoverState
 
+	let syncManager = AppDelegate.get().syncManager!
+	
+	@ViewBuilder
 	var body: some View {
-		let status = connectionsMonitor.connections.global
 		
-		Group {
-			Button {
-				showConnectionsPopover()
-			} label: {
-				HStack {
-					Image("ic_connection_lost")
-						.resizable()
-						.frame(width: 16, height: 16)
-					Text(status.localizedText())
+		Button {
+			showAppStatusPopover()
+		} label: {
+			buttonContent
+		}
+		.buttonStyle(PlainButtonStyle())
+		.padding(.all, 7)
+		.background(Color.buttonFill)
+		.cornerRadius(30)
+		.overlay(
+			RoundedRectangle(cornerRadius: 30) // Test this with larger dynamicFontSize
+				.stroke(Color.borderColor, lineWidth: 1)
+		)
+		.onReceive(syncManager.statePublisher) {
+			syncManagerStateChanged($0)
+		}
+		.onReceive(syncManager.pendingSettingsPublisher) {
+			syncManagerPendingSettingsChanged($0)
+		}
+	}
+	
+	@ViewBuilder
+	var buttonContent: some View {
+		
+		let connectionStatus = connectionsMonitor.connections.global
+		if connectionStatus == .closed {
+			HStack(alignment: .firstTextBaseline, spacing: 3) {
+				Image(systemName: "bolt.slash.fill")
+					.imageScale(.large)
+				Text(NSLocalizedString("Offline", comment: "Connection state"))
+					.padding(.trailing, 6)
+			}
+			.font(.caption2)
+		}
+		else if connectionStatus == .establishing {
+			HStack(alignment: .firstTextBaseline, spacing: 3) {
+				Image(systemName: "bolt.slash")
+					.imageScale(.large)
+				Text(NSLocalizedString("Connecting...", comment: "Connection state"))
+					.padding(.trailing, 6)
+			}
+			.font(.caption2)
+		} else /* .established */ {
+			
+			if pendingSettings != nil {
+				// The user enabled/disabled cloud sync.
+				// We are using a 30 second delay before we start operating on the user's decision.
+				// Just in-case it was an accidental change, or the user changes his/her mind.
```

Comment:
It's a nice feature but the way it is displayed is a bit confusing. Instead of showing a 30s timer with a skip button, we could show: `Cloud sync will be disabled (or enabled)` with a `Cancel`. The time is still there but not visible. I think it makes a bit more sense from the user's point of view.

Reply:
Done in b02fba0

<img width="250" alt="Screen Shot 2021-08-19 at 11 59 58" src="https://user-images.githubusercontent.com/304604/130112170-b2d8ef9c-412f-4e2c-87e4-0670bae2c7e1.png">